### PR TITLE
Try to get .env filename from env first

### PR DESCRIPTION
**Pull request by cyberplant**
Inception warning!! 😄 

TL;DR: this patch allows us to do: 
`DOTENV_FILENAME=/directory/.env godotenv /bin/application`
or configure `DOTENV_FILENAME` as env var for an app that has autoload enabled.

Long explanation:
We're using `drone` for CI, and the latest versions are autoloading env vars using godotenv. The problem is that in our setup (mesos/marathon) we cannot put files wherever we want, but we can configure environment variables!

So with this little change we can run the drone image, put the config file in a specific dir and by using an env variable, make drone load it using godotenv.

I know you don't want to have much differences from the original dotenv.. If this is much of a problem, I can make a PR for them too. I don't know much about ruby, but don't know much about go as well!

Regards!

### CODE REVIEW

**Comment by joho** [comment]
Wouldn't this be better treated as an application level concern as it sounds very specific to your architecture?

I'm just coding in the editor (so this may not compile), but why not something like this in your code rather than pushing down into the library?

```go
envFileLocation, present := os.LookupEnv("DOTENV_FILENAME")
if present {
    godotenv.Load(envFileLocation)
} else {
    godotenv.Load()
}
```

**Comment by cyberplant** [comment]
it's not my app.. I can do a PR on drone, but they will lose the autoload feature that is pretty handy. 

Also I think other people can also benefit from this, and I don't think of much drawbacks.

I'm not sure about my implementation.. it works, but maybe there are better ways to do it (go newbie).

**Comment by cyberplant** [comment]
What do you think @joho ?

I have some options here: 
- I can submit a PR on upstream dotenv project for this same feature, if this will encourage you to merge this PR. I prefer this option as I understand is something that should be solved on the library itself. Also, I don't like much to have 'hardcoded' things, maybe that's why I'm biased towards this change.
 
- Submit a PR for drone to disable the autoload feature and manually load env.

**Comment by joho** [comment]
@cyberplant the code snippet I shared will still autoload env if the specific env var specifying a path is not present, so I don't think of it disabling the autoload.

Having thought on this further I don't think this PR suits this repository or the upstream dotenv one. The library's primary purpose is to provide a streamlined way to work with [12 factor](https://12factor.net/) applications in a dev environment. In CI & production the app should probably be using a different technique for injecting environment vars.

My recommendation is to _probably_ change your CI to set the ENV vars you need directly rather than relying on a `.env` file in that environment, and failing that to try and change the drone repository. I'm going to close this out now, but thanks so much for taking the time to prep a PR and actively seek feedback 👍

**Comment by cyberplant** [comment]
I cannot do what you suggested as this will expose sensitive data as credentials, we need to have that in a file that is going to be loaded afterwards from a different place.

Anyway, thanks for your action, I know now that I'll have to find another ways to achieve it.
